backend/de.metas.business/src/main/java/de/metas/invoice/service/impl/AbstractInvoiceBL.java

```diff
@@ -538,14 +540,28 @@ public final boolean testAllocation(final org.compiere.model.I_C_Invoice invoice
 				total = total.negate();
 			}
 
-			final boolean test = total.compareTo(alloc) == 0;
-			change = test != invoice.isPaid();
+			final BigDecimal openAmt = total.subtract(alloc);
+			final boolean fullyPaid = openAmt.signum() == 0;
+			change = fullyPaid != invoice.isPaid();
 			if (change)
 			{
-				invoice.setIsPaid(test);
+				invoice.setIsPaid(fullyPaid);
 			}
 
-			log.debug("IsPaid={} (allocated={}, invoiceGrandTotal={})", test, alloc, total);
+			final boolean isPartiallyPaid = !fullyPaid && hasAllocations;
+			change = isPartiallyPaid != invoice.isPartiallyPaid();
+			if (change)
+			{
+				invoice.setIsPartiallyPaid(isPartiallyPaid);
+			}
+			
+			change = openAmt.compareTo(invoice.getOpenAmt()) != 0;
```

Comment:
@metas-ts  @dm-ps IMHO, reusing the change variable is not correct
consider the case when fullyPaid changes but partial paid is not changed.

you will return false!  ... but true shall be returned